app/bundles/LeadBundle/EventListener/DashboardSubscriber.php

```diff
@@ -81,21 +81,193 @@ public function onWidgetDetailGenerate(WidgetDetailEvent $event): void
             }
 
             if (!$event->isCached()) {
+                $chartData = $this->leadModel->getLeadsLineChartData(
```

Comment:
I suggest breaking the long method into smaller methods; this will improve the DX and help with maintainability.

Reply:
Do you suggest converting all the changes from line 84 to 268 into smaller helper methods?